lib/src/module/built_in.dart

```diff
@@ -17,19 +17,23 @@ class BuiltInModule<T extends AsyncCallable> implements Module<T> {
   final Uri url;
   final Map<String, T> functions;
   final Map<String, T> mixins;
+  final Map<String, Value> variables;
 
   List<Module<T>> get upstream => const [];
-  Map<String, Value> get variables => const {};
   Map<String, AstNode> get variableNodes => const {};
   Extender get extender => Extender.empty;
   CssStylesheet get css => CssStylesheet.empty(url: url);
   bool get transitivelyContainsCss => false;
   bool get transitivelyContainsExtensions => false;
 
-  BuiltInModule(String name, {Iterable<T> functions, Iterable<T> mixins})
+  BuiltInModule(String name,
+      {Iterable<T> functions, Iterable<T> mixins, Map<String, Value> variables})
       : url = Uri(scheme: "sass", path: name),
         functions = _callableMap(functions),
-        mixins = _callableMap(mixins);
+        mixins = _callableMap(mixins),
+        variables = variables == null
+            ? UnmodifiableMapView({})
```

Comment:
```suggestion
            ? const {}
```

A constant map is implicitly unmodifiable, and it'll save some allocations.

Reply:
Thanks. Javascript habits are hard to shake ;)